### PR TITLE
SRCH-3206 Update attribute serialization and undo yaml unsafe load

### PR DESCRIPTION
**Pull request by lorawoodford**
## Summary
- Allows for the removal of `config.active_record.use_yaml_unsafe_load = true` from `application.rb` by leveraging built in JSON data types (available since Rails 5/MySQL 5.7) for the following:
    - `properties` attribute of `news_items`;
    - `urls` attribute of `tweets`;
    - `member_ids` attribute of `twitter_lists`;
    - `conditions` attribute of `watcher`; and, 
    - `news_item_ids` attribute of `youtube_playlists`.
- For the watcher and news item models where some existing behavior leaned on the predictability of the formerly serialized hashes, I introduced `attr_json` to simplify this process.
- After `bundle update`ing, webmock started throwing warnings about our pinned version of `curb` (this can be seen on main after running `bundle update` now).  Though ancillary, I've addressed this here as well.
- Many, many rubocop updates have been pushed in the second commit (hopefully making it simpler to do code review of the primary code changes).
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks

- [x] You have run `bundle update` and committed your changes to Gemfile.lock.
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
 
:warning: You have squashed your commits into a single commit (exceptions: your PR includes commits with formatting-only changes, such as required by Rubocop or Cookstyle, or if this is a feature branch that includes multiple commits).
I've intentionally sequestered all Rubocop only changes (there are many) into a secondary commit to hopefully make code review simpler.
 
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
:warning: Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
While 117 pre-existing Code Climate issues are fixed in this PR, 4 less trivial errors remain.
 
#### Process Checks

- [X] You have specified at least one "Reviewer".

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ gem 'rack-contrib', '~> 2.1.0'
 gem 'rails-observers', '~> 0.1.5'
 gem 'responders', '~> 3.0.1'
 gem 'mysql2', '~> 0.5.0'
-gem 'curb', '~> 0.9.4'
+gem 'curb', '~> 1.0', '>= 1.0.1'
```
Comment (MothOnMars) [inline Gemfile:10]: Let's just use: `'~> 1.0.1'`. I don't trust `curb` enough to bump minor versions with a `bundle update`. Eventually I would like to remove `curb` entirely in favor of either `faraday` or `http`.

**Review by MothOnMars — CHANGES_REQUESTED** [review]
Per our discussion, we will simplify this code by:
- removing the new gem
- backing out the db changes
- using ActiveRecord::Store

POC of the simplified changeset:
https://github.com/GSA/search-gov/compare/main...MothOnMars:unsafe_3206?expand=1

We'll also add comments alongside the `store` methods indicating that in the future, we may want to change the db column type to `json`.

**Comment by lorawoodford** [comment]
@MothOnMars Changes made.  Please re-review when able.  Thanks!

File: spec/lib/importers/rss_feed_data_spec.rb [hunk 2]
```diff
@@ -31,23 +35,23 @@
...
+              '200 OK'
+            end
           { arg => status_code }
         end
         rss_feed_url.news_items.destroy_all
         described_class.new(rss_feed_url).import
       end
 
-      it 'should populate and index just the news items that are valid' do
+      it 'populates and index just the news items that are valid' do
```
Comment (MothOnMars) [inline spec/lib/importers/rss_feed_data_spec.rb:50]: Should be "populates and indexes". 

It's a symptom of a spec that's testing too many things, but unfortunately that's the case for a lot of our older specs. I'm going to create a ticket for one of the new folks to disable auto-correction for that cop, as it has tripped us up in various ways in the past.

See also https://github.com/rubocop/rubocop-rspec/issues/1179

File: spec/lib/importers/rss_feed_data_spec.rb [hunk 3]
```diff
@@ -63,11 +67,11 @@
...
```
Comment (MothOnMars) [inline spec/lib/importers/rss_feed_data_spec.rb:70]: Should be "populates and indexes"

File: spec/lib/importers/rss_feed_data_spec.rb [hunk 4]
```diff
@@ -76,17 +80,18 @@
...
       end
+
       before do
         stub_request(:get, rss_feed_url.url).to_return({ status: 200, body: rss_feed_content })
         rss_feed_url.news_items.destroy_all
         described_class.new(rss_feed_url).import
       end
 
-      it 'should populate and index just the news items that are valid' do
+      it 'populates and index just the news items that are valid' do
```
Comment (MothOnMars) [inline spec/lib/importers/rss_feed_data_spec.rb:90]: Should be "populates and indexes"

File: spec/lib/importers/rss_feed_data_spec.rb [hunk 5]
```diff
@@ -222,7 +230,7 @@
         expect(rss_feed_url).to receive(:touch).with(:last_crawled_at).and_raise StandardError.new('Error Message!')
       end
 
-      it 'should log it and move on' do
+      it 'logs it and move on' do
```
Comment (MothOnMars) [inline spec/lib/importers/rss_feed_data_spec.rb:233]: Should be "logs it and moves on"

File: spec/lib/importers/rss_feed_data_spec.rb [hunk 6]
```diff
@@ -402,7 +415,7 @@
         stub_request(:get, rss_feed_url.url).to_return({ status: 200, body: rss_feed_content })
       end
 
-      it 'should not change the number of news items, and update the crawl status' do
+      it 'does not change the number of news items, and update the crawl status' do
```
Comment (MothOnMars) [inline spec/lib/importers/rss_feed_data_spec.rb:418]: Should be "does not change the number of news items, and updates the crawl status"

File: spec/lib/tasks/twitter_spec.rb [hunk 7]
```diff
@@ -218,34 +221,34 @@
           expect(Tweet.count).to eq(0)
         end
 
-        it 'should only create a new tweet if the user id matches a TwitterProfile' do
+        it 'onlies create a new tweet if the user id matches a TwitterProfile' do
```
Comment (MothOnMars) [inline spec/lib/tasks/twitter_spec.rb:224]: Should be "only creates"

**Review by MothOnMars — CHANGES_REQUESTED** [review]
Code looks good. Just need to tweak some of Rubocop's autocorrections.

**Comment by lorawoodford** [comment]
@MothOnMars Requested changes made.

**Review by MothOnMars — DISMISSED** [review]
LGTM

**Comment by MothOnMars** [comment]
SRCH-3206 has been closed as done. Additional changes will be made to this PR to support a "safe" migration via https://cm-jira.usa.gov/browse/SRCH-3345, which will also cover the deployment of these changes.

**Comment by MothOnMars** [comment]
Closing this up. I'm tweaking these changes for SRCH-3345 and breaking this into smaller chunks to simplify testing & deployment.

**Pull request closed without merging**